### ISSUE TITLE
json.SyntaxError on push

### ISSUE THREAD

**Issue by KatalystGames** [itchio/butler#95]
My project has randomly started getting the below error when attempting to push to itch.

> *json.SyntaxError invalid character '<' looking for beginning of value
C:/Multi-Runner/builds/58469115/0/itchio/butler/src/github.com/itchio/butler/push.go:29 (0x414c97)
C:/Multi-Runner/builds/58469115/0/itchio/butler/src/github.com/itchio/butler/main.go:475 (0x41c2fd)
C:/Go/src/runtime/proc.go:183 (0x450a0e)
C:/Go/src/runtime/asm_386.s:1612 (0x47a831)

**Comment by fasterthanlime** [comment]
Is this still happening? I'm not able to reproduce, might have been a fluke of the API server

**Comment by KatalystGames** [comment]
Yes, it is still happening. I just tried to push a different project I have on Itch and it is also getting that same error now.

> C:\Users\Michael>butler push D:\UnityWork\Builds\SciFiTowerDefense\Windows katalystkreations/xenostorm:windows-alpha --userversion-file D:\UnityWork\Builds\SciFiTowerDefense\version.txt
*json.SyntaxError invalid character '<' looking for beginning of value
C:/Multi-Runner/builds/58469115/0/itchio/butler/src/github.com/itchio/butler/push.go:29 (0x414c97)
C:/Multi-Runner/builds/58469115/0/itchio/butler/src/github.com/itchio/butler/main.go:475 (0x41c2fd)
C:/Go/src/runtime/proc.go:183 (0x450a0e)
C:/Go/src/runtime/asm_386.s:1612 (0x47a831)

**Comment by fasterthanlime** [comment]
Are you using a proxy? Which country are you pushing from?

**Comment by KatalystGames** [comment]
No proxy and I'm from the US.

**Comment by leafo** [comment]
I just pushed a copy of your current windows build to your project page to test, and did not encounter any issues. Do other commands work as expected, can you run the command `butler status katalystkreations/xenostorm:windows-alpha`?

**Comment by KatalystGames** [comment]
Yup, the rest of the butler commands work fine. Just when I push.

> C:\Users\Michael>butler status katalystkreations/xenostorm:windows-alpha
+---------------+---------+------------------------+--------------------------------+
|    CHANNEL    | UPLOAD  |         BUILD          |             STATE              |
+---------------+---------+------------------------+--------------------------------+
| windows-alpha | #334802 | √ #12408 (from #12073) | √ 477 KiB signature, √ 2.8 KiB |
|               |         |                        | patch, √ 423 MiB archive       |
+---------------+---------+------------------------+--------------------------------+

**Comment by fasterthanlime** [comment]
@KatalystGames can you post the output of the same push command but with the `-v` flag added? No worries, we'll get to the bottom of this :) I'll make a build with more detailed output soon so we can see what's really going on.

**Comment by KatalystGames** [comment]
> C:\Users\Michael>butler -v push D:\UnityWork\Builds\SciFiTowerDefense katalystkreations/xenostorm:windows-alpha --userversion-file D:\UnityWork\Builds\SciFiTowerDefense\version.txt
*json.SyntaxError invalid character '<' looking for beginning of value
C:/Multi-Runner/builds/58469115/0/itchio/butler/src/github.com/itchio/butler/push.go:29 (0x415c1f)
C:/Multi-Runner/builds/58469115/0/itchio/butler/src/github.com/itchio/butler/main.go:475 (0x41dade)
C:/Go/src/runtime/proc.go:183 (0x453fb0)
C:/Go/src/runtime/asm_amd64.s:2086 (0x482fd1)

**Comment by fasterthanlime** [comment]
Can you update butler to the latest version (`butler upgrade --head`) and try again? The message should be more detailed this time.

**Comment by KatalystGames** [comment]
> C:\Users\Michael>butler -v push D:\UnityWork\Builds\SciFiTowerDefense katalystkreations/xenostorm:windows-alpha --userversion-file D:\UnityWork\Builds\SciFiTowerDefense\version.txt
*json.SyntaxError invalid character '<' looking for beginning of value
C:/Multi-Runner/builds/58469115/0/itchio/butler/src/github.com/itchio/butler/vendor/github.com/itchio/go-itchio/itchio.go:189 (0x59ad42)
C:/Multi-Runner/builds/58469115/0/itchio/butler/src/github.com/itchio/butler/push.go:53 (0x415edd)
C:/Multi-Runner/builds/58469115/0/itchio/butler/src/github.com/itchio/butler/push.go:29 (0x415c1f)
C:/Multi-Runner/builds/58469115/0/itchio/butler/src/github.com/itchio/butler/main.go:475 (0x41dade)
C:/Go/src/runtime/proc.go:183 (0x453fb0)
C:/Go/src/runtime/asm_amd64.s:2086 (0x482fd1)

**Comment by fasterthanlime** [comment]
I've made it print the value it can't parse - can you upgrade and try again? same as before.

**Comment by KatalystGames** [comment]
Due to the div tags it was causing havoc to post so here is a screenshot! lol
https://gyazo.com/e8bdccd448ea35dfd5292b4973f5b1ef

**Comment by fasterthanlime** [comment]
Thanks! Can you upload the contents of `D:\UnityWork\Builds\SciFiTowerDefense\version.txt` as a github attachment?

**Comment by KatalystGames** [comment]
[version.txt](https://github.com/itchio/butler/files/659137/version.txt)

**Comment by fasterthanlime** [comment]
Actually, I'm not! That's the problem :) it begins with a UTF-8 BOM (FF, FE) - I'll need to strip that in butler :)

**Comment by KatalystGames** [comment]
Ahh, must be the way my Unity Editor script was generating text files that differed from when I manually did it with Windows previously. :D

**Comment by KatalystGames** [comment]
Which formats are supported so I can make sure when it creates the file, its using one of those?

**Comment by fasterthanlime** [comment]
UTF-8 is the preferred standard. I hadn't seen UTF-16 in a while! (except when dealing with windows formats).

That said I'd rather modify butler so that it accepts UTF-16 as well, just so anyone runs into the same issue!

**Comment by dw5** [comment]
I got this, due to having somewhere charset=us-ascii

```
• Pushing 702.37 MiB (7036 files, 264 dirs, 0 symlinks)
Building diff context
string field contains invalid UTF-8         
```

Maybe charset converter should/could be built in to butler? Like if it detects, yes no pops up, and also be available as additional utility command?
